packages/s2-core/src/facet/layout/build-row-tree-hierarchy.ts

```diff
@@ -39,11 +40,16 @@ export const buildRowTreeHierarchy = (params: TreeHeaderParams) => {
   const isDrillDownItem = spreadsheet.dataCfg.fields.rows?.length <= level;
   const sortedDimensionValues =
     (dataSet as PivotDataSet)?.sortedDimensionValues?.[currentField] || [];
-
+  const dimensions =
+    ROOT_ID === id
+      ? sortedDimensionValues
+      : sortedDimensionValues?.filter((item) =>
+          item?.includes(id?.split(`${ROOT_ID}${ID_SEPARATOR}`)[1]),
```

Comment:
虽然好像也差不多， replace 呢？

```suggestion
          item?.includes(id?.replace(`${ROOT_ID}${ID_SEPARATOR}`, ''))
```